sys/kern/subr_intr.c

```diff
@@ -479,6 +478,23 @@ isrc_free_irq(struct intr_irqsrc *isrc)
 	return (0);
 }
 
+/*
+ *  Destroy interrupt event for interrupt source.
+ */
+static int
+isrc_event_destroy(struct intr_irqsrc *isrc)
+{
+	int rc;
+
+	MPASS(isrc->isrc_irq >= intr_nirq);
+
```

Comment:
So this is backwards. MPASS(ex...) means ex must be true. However, this can never be true (since IRQs are necessarily smaller than intr_nirq. So this will always panic. Again, how did you test this?

Reply:
Notice another function called near this function?  What does that function do?  Having considered the situation, I do believe it is appropriate for this one to be called second.